### PR TITLE
Internal: Nested Tabs markup tweaks round 2 [ED-11575]

### PR DESCRIPTION
**Pull request by hein-obox**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/nested-tabs-html/assets/js/frontend/handlers/nested-tabs-html.js [hunk 1]
```diff
@@ -142,15 +146,15 @@ export default class NestedTabsHtml extends Base {
 		}
 	}
 
-	deactivateActiveTab( tabIndex ) {
+	deactivateActiveTab( tabIndex = 0, eventType ) {
```
Comment (hein-obox) [inline modules/nested-tabs-html/assets/js/frontend/handlers/nested-tabs-html.js:149]: Small tweak used for the deactivation of menu titles on hover.

File: modules/nested-tabs-html/assets/js/frontend/handlers/nested-tabs-html.js [hunk 2]
```diff
@@ -531,21 +535,21 @@ export default class NestedTabsHtml extends Base {
...
-		if ( elementorFrontend.isEditMode() ) {
+		const widgetSelector = this.getSettings( 'selectors' ).widgetContainer;
+
+		if ( elementorFrontend.isEditMode() || 'resize' === event?.type ) {
 			const responsiveDevices = [ 'mobile', 'mobile_extra', 'tablet', 'tablet_extra' ],
 				currentDevice = elementorFrontend.getCurrentDeviceMode();
 
 			if ( -1 !== responsiveDevices.indexOf( currentDevice ) ) {
-				this.$element.find( '.e-n-tabs' ).attr( 'data-touch-mode', 'true' );
+				this.$element.find( widgetSelector ).attr( 'data-touch-mode', 'true' );
```
Comment (hein-obox) [inline modules/nested-tabs-html/assets/js/frontend/handlers/nested-tabs-html.js:545]: I am using the same function for the mega menu and therefore I needed dynamic selectors.

File: modules/nested-tabs-html/widgets/nested-tabs-html.php [hunk 3]
```diff
@@ -1123,24 +1122,41 @@ protected function render_tab_titles_html( $item_settings ): string {
...
-		$render_attributes = $active_class . $this->get_render_attribute_string( $setting_key ) . str_replace( '"', '', $attribute_selector );
 
-		return str_replace( $attribute_selector, $render_attributes, $container_html );
+	/**
+	 * Print the content area.
+	 *
+	 * @param int $index
+	 * @param array $item_settings
+	 */
+	public function print_child( $index, $item_settings = [] ) {
```
Comment (hein-obox) [inline modules/nested-tabs-html/widgets/nested-tabs-html.php:1141]: I have copied the existing approach from the Mega Menu. This approach is better than my previous approach.

File: modules/nested-tabs-html/assets/scss/frontend.scss [hunk 4]
```diff
@@ -189,6 +188,13 @@
 				-webkit-text-stroke-width var( --n-tabs-title-transition ),
 				-webkit-text-stroke-color var( --n-tabs-title-transition ),
 				transform var( --n-tabs-title-transition );
+			-webkit-tap-highlight-color: transparent;
```
Comment (hein-obox) [inline modules/nested-tabs-html/assets/scss/frontend.scss:191]: Old styling that I had forgotten in my previous update.

File: modules/nested-tabs-html/assets/scss/frontend.scss [hunk 5]
```diff
@@ -190,6 +189,15 @@
...
 				transform var( --n-tabs-title-transition );
 
+			// Remove tap highlight color on iOS: https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-tap-highlight-color.
+			-webkit-tap-highlight-color: transparent;
+			-webkit-touch-callout: none;
+			-webkit-user-select: none;
+			-khtml-user-select: none;
+			-moz-user-select: none;
+			-ms-user-select: none;
+			user-select: none;
```
Comment (rami-elementor) [inline modules/nested-tabs-html/assets/scss/frontend.scss:199]: Do we need all the browser prefixes?

I think that the `autoprefixer` script adds them based on our `.browserslistrc` settings.

File: modules/nested-tabs-html/widgets/nested-tabs-html.php [hunk 6]
```diff
@@ -1123,24 +1122,41 @@ protected function render_tab_titles_html( $item_settings ): string {
...
 		ob_start();
-		$this->print_child( $item_settings['index'] );
+		$this->print_child( $item_settings['index'], $item_settings );
 		$container_html = ob_get_clean();
 
-		$setting_key = $this->get_repeater_setting_key( 'tab_content', 'tabs', $item_settings['index'] );
-		$attribute_selector = '" data-id=';
-		$active_class = 0 === $item_settings['index'] ? ' e-active"' : '"';
+		$attribute_selector = 'class="';
+		$active_class = 0 === $item_settings['index'] ? $attribute_selector . 'e-active ' : $attribute_selector;
```
Comment (davseve) [inline modules/nested-tabs-html/widgets/nested-tabs-html.php:1129]: I see that there are some cases that the class attr will contain no values E.g, `class=""`.
In those cases, do we still need to add an empty class attr?

**Review by davseve — COMMENTED** [review]
Do we need to add new tests for this task or that the existing tests cover the new code tweaks?
Do we need a mind map?

File: modules/nested-tabs-html/assets/scss/frontend.scss (hunk 5 above)
Comment (hein-obox) [reply modules/nested-tabs-html/assets/scss/frontend.scss:199]: @rami-elementor You are correct. I have removed them.

File: modules/nested-tabs-html/widgets/nested-tabs-html.php (hunk 6 above)
Comment (hein-obox) [reply modules/nested-tabs-html/widgets/nested-tabs-html.php:1129]: Updated. I have made the adding of classes conditional.

**Comment by hein-obox** [comment]
@davseve This is covered by the existing tests.

**Review by davseve — APPROVED** [review]

**Review by louiswol94 — APPROVED** [review]

**Pull request merged**